clients/android/app/src/main/java/app/lockbook/utils/Converters.kt

```diff
@@ -22,9 +109,35 @@ val createAccountConverter = object : Converter {
     override fun canConvert(cls: Class<*>): Boolean = true
 
     override fun fromJson(jv: JsonValue): Any? = when (jv.obj?.string("tag")) {
-        "Ok" -> Ok(Unit)
-        "Err" -> matchError(jv)
-        else -> Err(CoreError.Unexpected("Unable to parse tag: ${jv.obj?.toJsonString()}"))
+        okTag -> Ok(Unit)
+        errTag -> when (jv.obj?.obj("content")?.string("tag")) {
+            uiErrorTag -> {
+                val error = jv.obj?.obj("content")?.string("content")
+                if (error != null) {
+                    Err(
+                        when (error) {
+                            CreateAccountError.UsernameTaken::class.simpleName -> CreateAccountError.UsernameTaken
+                            CreateAccountError.InvalidUsername::class.simpleName -> CreateAccountError.InvalidUsername
+                            CreateAccountError.CouldNotReachServer::class.simpleName -> CreateAccountError.CouldNotReachServer
+                            CreateAccountError.AccountExistsAlready::class.simpleName -> CreateAccountError.AccountExistsAlready
+                            else -> CreateAccountError.Unexpected("Can't recognize UiError content.")
```

Comment:
Make error more specific enough to allow a person to find out where exactly it happened.